agents/logsuploader/main.go

```diff
@@ -1,27 +1,105 @@
 package main
 
 import (
+	"errors"
 	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	"github.com/mitchellh/go-ps"
 
 	"github.com/microsoft/ApplicationInsights-Go/appinsights"
 )
 
 func main() {
-	fmt.Println("Hello, logs uploader.")
-	var name string
-	client := appinsights.NewTelemetryClient("320dcf98-173f-429b-ab39-df8b4951fb94")
-	metric := appinsights.NewMetricTelemetry("Queue length", 1)
-	event := appinsights.NewEventTelemetry("event")
-	metric.Properties["Queue name"] = "queuename"
-	client.Track(metric)
-	client.Track(event)
+	fmt.Println("Starting logs uploader.")
+
+	if checkFirstInstance() == false {
```

Comment:
Can we not quit, but rather sleep in a loop and check if it's running every couple of minutes, and start it if it failed?

Reply:
we could , i want to minimize resource use , thus exit so we only have one instance up and running

Reply:
I'd say do it. As long as it only wakes up maybe once a minute - it's worth it. And if possible to log from here - log an error message saying the spawned process disappeared, that way we'll know if this is a persistent problem.